connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/serializers/RowDataTest.java

```diff
@@ -0,0 +1,469 @@
+                        DataTypes.FIELD(TestingUtils.ROW_KEY_FIELD, DataTypes.STRING()),
+                        DataTypes.FIELD(TestingUtils.NESTED_COLUMN_FAMILY_1, outerNest));
+
+        Assertions.assertThatThrownBy(
+                        () ->
+                                RowDataToRowMutationSerializer.builder()
+                                        .withSchema(dtDoubleNestedSchema)
+                                        .withRowKeyField(TestingUtils.ROW_KEY_FIELD)
+                                        .withNestedRowsMode()
+                                        .build())
+                .hasMessageContaining(ErrorMessages.NESTED_TYPE_ERROR);
+    }
+
+    @Test
+    public void testTimestampPrecisionError() {
+        DataType dtTimestampSchema =
+                DataTypes.ROW(
+                        DataTypes.FIELD(TestingUtils.ROW_KEY_FIELD, DataTypes.STRING()),
+                        DataTypes.FIELD(
+                                "timestamp",
+                                DataTypes.TIMESTAMP(
+                                        RowDataToRowMutationSerializer.MAX_DATETIME_PRECISION
+                                                + 1)));
+        Assertions.assertThatThrownBy(
+                        () ->
+                                RowDataToRowMutationSerializer.builder()
+                                        .withSchema(dtTimestampSchema)
+                                        .withRowKeyField(TestingUtils.ROW_KEY_FIELD)
+                                        .withColumnFamily(TestingUtils.COLUMN_FAMILY)
+                                        .build())
+                .hasMessageContaining(
+                        String.format(
+                                ErrorMessages.TIMESTAMP_OUTSIDE_PRECISION_TEMPLATE,
+                                DataTypes.TIMESTAMP(
+                                        RowDataToRowMutationSerializer.MAX_DATETIME_PRECISION + 1),
+                                RowDataToRowMutationSerializer.MIN_DATETIME_PRECISION,
+                                RowDataToRowMutationSerializer.MAX_DATETIME_PRECISION));
+    }
+
+    @Test
+    public void testRowKeyNoStringerror() {
+        DataType dtIntegerKeySchema =
+                DataTypes.ROW(DataTypes.FIELD(TestingUtils.ROW_KEY_FIELD, DataTypes.INT()));
+
+        Assertions.assertThatThrownBy(
+                        () ->
+                                RowDataToRowMutationSerializer.builder()
+                                        .withSchema(dtIntegerKeySchema)
+                                        .withRowKeyField(TestingUtils.ROW_KEY_FIELD)
+                                        .withColumnFamily(TestingUtils.COLUMN_FAMILY)
+                                        .build())
+                .hasMessage(
+                        String.format(ErrorMessages.ROW_KEY_STRING_TYPE_TEMPLATE, DataTypes.INT()));
+    }
+
+    @Test
+    public void testMissingRowKey() {
```

Comment:
out fo curiosity, what's the behavior if the rowkey is a field in a nested row?

Reply:
It will fail for not being a string